core/src/main.cpp

```diff
@@ -363,20 +364,33 @@ auto load_oidc_configuration(const json& _config, json& _oi_config, json& _endpo
 		beast::tcp_stream tcp_stream{io_ctx};
 
 		// Load config
-		const auto host{_oi_config.at("config_host").get_ref<const std::string&>()};
-		const auto port{std::to_string(_oi_config.at("port").get<int>())};
-		const auto temp_uri{_oi_config.at("well_known_uri").get_ref<const std::string&>()};
-		const auto uri{fmt::format("{}/.well-known/openid-configuration", temp_uri)};
+		const auto provider{_oi_config.at("provider_url").get_ref<const std::string&>()};
+		auto parsed_uri{boost::urls::parse_uri(provider)};
+
+		if (parsed_uri.has_error()) {
+			log::error("{}: Error trying to parse 'provider_url'. Please check configuration.", __func__);
+			return false;
+		}
+
+		auto url{*parsed_uri};
+		auto host{url.host()};
```

Comment:
Are copies intended here?

Could be marked `const`.

Reply:
`const`ed any new variables I could.